FSharpLu.Json/Compact.fs

```diff
@@ -123,10 +123,18 @@ type CompactUnionJsonConverter(?tupleAsHeterogneousArray:bool) =
                         // and serializing the nested object directly
                         serializer.Serialize(writer, innerValue)
         // Tuple
-        else if tupleAsHeterogneousArray && isTupleType t then
-            let v = FSharpValue.GetTupleFields value
-            serializer.Serialize(writer, v)
-
+        else if isTupleType t then
+            if tupleAsHeterogneousArray then
+                let v = FSharpValue.GetTupleFields value
+                serializer.Serialize(writer, v)
+            else
+                // need to serialize tuple as object format, but can't just call serializer.Serialize(writer, value) or will infinitely recur
```

Comment:
Can we instead fall-back on the base class implementation of `WriteJson` (from `Newtonsoft.Json.JsonConverter`) to perform the serialization of the tuple as an JSON object? By just calling `base.WriteJson(...)`?

Reply:
I fixed `CanConvert` in my branch: after rebasing you should be able to remove this `else` branch altogether.